### PR TITLE
Client work

### PR DESCRIPTION
**Pull request by shrugs**
creating a PR so I can make comments, but otherwise ignore this

### CODE REVIEW

File: bridge-to-the-moon/bttm-cli.js [hunk 1]
```diff
@@ -0,0 +1,62 @@
+require('dotenv').config()
+const vorpal = require('vorpal')()
+const asDefault = require('vorpal-as-default')
+const Web3 = require('web3')
+const selfText = require('./src/utils/selfText')
+const setup = require('./src/utils/setup')
+
+// let's monekypatch web3 cause it doesn't play nice with truffle yet
```
Comment (shrugs) [inline bridge-to-the-moon/bttm-cli.js]: remove if we don't need to monkeypatch the asyncSend funciton anymore

File: bridge-to-the-moon/src/offchain/deploy.js [hunk 2]
```diff
@@ -0,0 +1,96 @@
...
+const spawn = require('tape-spawn')
+const net = require('net')
+const Web3 = require('web3')
+const process = require('process')
+const color = require('colors')
+const child_process = require('child_process')
+
+const useLocalSolc = true
+
+const web3 = new Web3(new Web3.providers.HttpProvider("http://localhost:4242"))
```
Comment (shrugs) [inline bridge-to-the-moon/src/offchain/deploy.js]: replace with ENV var

File: bridge-to-the-moon/src/offchain/deploy.js [hunk 3]
```diff
@@ -0,0 +1,96 @@
+const fs = require('fs')
```
Comment (shrugs) [inline bridge-to-the-moon/src/offchain/deploy.js]: why are we writing our own deploy script with compilation, etc?

File: bridge-to-the-moon/src/offchain/offchain.js [hunk 4]
```diff
@@ -0,0 +1,61 @@
+const Web3 = require('web3');
+const fs = require('fs');
+const deploy = require('./deploy');
+
+const web3 = new Web3(new Web3.providers.HttpProvider("http://localhost:4242"))
```
Comment (shrugs) [inline bridge-to-the-moon/src/offchain/offchain.js]: ENV var, why create a new web3 when we have the factory?

**Review by shrugs — COMMENTED** [review]
What are the primary changes here? Have you been able to run it against the test contracts?

**Comment by shrugs** [comment]
cc @hswick

**Comment by hswick** [comment]
@shrugs biggest changes so far have been switching stuff to old version of web3

File: bridge-to-the-moon/src/offchain/deploy.js (hunk 3 above)
Comment (hswick) [reply bridge-to-the-moon/src/offchain/deploy.js]: That is for offchain deployment stuff. It is being deployed to the parity chain, there might be an easier way to do it but thats how it got it to work so far. The truffle stuff deploys stuff to onchain

**Comment by sinahab** [comment]
Merging so Oscar can begin integration with DogeRelay. We will push more updates to the client over the next days.

**Pull request merged**